cpp/tests/copying/concatenate_tests.cpp

```diff
@@ -59,8 +57,20 @@ struct TypedColumnTest : public cudf::test::BaseFixture {
   {
     auto typed_data = static_cast<char*>(data.data());
     auto typed_mask = static_cast<char*>(mask.data());
-    thrust::sequence(thrust::device, typed_data, typed_data + data_size());
-    thrust::sequence(thrust::device, typed_mask, typed_mask + mask_size());
+    std::vector<char> h_data(data_size());
+    thrust::sequence(h_data.begin(), h_data.end());
+    std::vector<char> h_mask(mask_size());
+    thrust::sequence(h_mask.begin(), h_mask.end());
+    CUDA_TRY(cudaMemcpyAsync(typed_data,
```

Comment:
I think you should either synchronize the stream right after these copies, or make them non-async `cudaMemcpy`.